xlcalculator/evaluator.py

```diff
@@ -27,10 +27,10 @@ def eval_cell(self, addr):
             raise RuntimeError(
                 f'Cycle detected for {addr}:\n- ' + '\n- '.join(self.seen))
         self.seen.append(addr)
+        self.sheet = addr.split("!")[0]
```

Comment:
There is an asymmetry here between where you set `sheet` and then reset it. This makes it very hard to see where you enter and leave a context. I would be nice to have this assignment and the reset() in the same function.

Reply:
Ideally proper contextmanager would be used in general. Unfortunately I don't have so much time to spend on it for now, it's more an idea of what would work than MR than should be pulled as is.

Reply:
@adanteh, I think a proper contextmanager is worthwhile. That said, not being able to cross-sheets is a fundamental problem. This proposal appears to resolve the cross-sheet problem for now. I have attempted to address the suggestions from @strichter and will update this PR with those changes shortly.

@adanteh, would this change adequately address your use case until we can raise a contextmanager?